### PR TITLE
mpw: Updated to 2.7-cli-1

### PR DESCRIPTION
**Pull request by lhunath**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/mpw.rb [hunk 1]
```diff
@@ -16,26 +16,24 @@ class Mpw < Formula
   end
 
   bottle do
-    sha256 cellar: :any, arm64_big_sur: "252fbad588409e4fd5be091f7c248150e75b995d39ce334938b9a7d33534cbc9"
-    sha256 cellar: :any, big_sur:       "4ae85b3d30e47294436b7fddca456c98ed2bf546793f2ef9d57a372d782fb072"
-    sha256 cellar: :any, catalina:      "2f275d762a9c73bd6b3f2e5a7f3f13a9c99ddfc3e2f89a2ededa07ba89b6de40"
-    sha256 cellar: :any, mojave:        "9103716223529cd3e2cb969e904892bf2022cb8e73918418f2d3d343d1325c80"
-    sha256 cellar: :any, high_sierra:   "07b89df8d96f9c1cebbf6296a4e98b2bac833c45f736b646a1eba24bd5244732"
+    cellar :any
```
Comment (SMillerDev) [inline Formula/mpw.rb:19]: The bottle block should not be edited.

File: Formula/mpw.rb [hunk 2]
```diff
@@ -16,26 +16,24 @@ class Mpw < Formula
...
 
-    ENV["targets"] = "mpw"
-    # not compatible with json-c 0.14 yet
-    ENV["mpw_json"] = "0"
+    ENV["targets"] = "mpw mpw-tests"
+    ENV["mpw_json"] = "1"
     ENV["mpw_color"] = "1"
 
     system "./build"
+    system "./mpw-tests"
```
Comment (SMillerDev) [inline Formula/mpw.rb:36]: Why are we running tests?

Comment (lhunath) [reply Formula/mpw.rb:36]: These are integration tests to ensure the build works as expected on the target machine, they are unavailable after install (ie. in the `test` block).

What is your recommendation?

Comment (SMillerDev) [reply Formula/mpw.rb:36]: Run integration tests during development and run a single command to check in the test block if homebrew didn't mess up the install.

File: Formula/mpw.rb [hunk 3]
```diff
@@ -16,26 +16,24 @@ class Mpw < Formula
...
     system "./mpw-cli-tests"
```
Comment (carlocab) [inline Formula/mpw.rb:37]: If these are just integration tests too (i.e. not necessary for the build) we should probably skip these also.

File: Formula/mpw.rb (hunk 2 above)
Comment (lhunath) [reply Formula/mpw.rb:36]: Is there a way for me to access testing assets, such as test cases, that are not part of the install, during `test`?

File: Formula/mpw.rb [hunk 4]
```diff
@@ -23,26 +23,174 @@ class Mpw < Formula
...
 
     system "./build"
-    system "./mpw-cli-tests"
     bin.install "mpw"
   end
 
   test do
-    assert_equal "Jejr5[RepuSosp",
-      shell_output("#{bin}/mpw -q -Fnone -u 'Robert Lee Mitchell' -M 'banana colored duckling' " \
-                   "-tlong -c1 -a3 'masterpasswordapp.com'").strip
```
Comment (carlocab) [inline Formula/mpw.rb:46]: Why do we need to replace this test?

Also, your replacement test has many style errors. The errors start [here](https://github.com/Homebrew/homebrew-core/pull/71428/checks?check_run_id=1929426297#step:5:17) and go on for a bit.

Comment (lhunath) [reply Formula/mpw.rb:46]: >_< – I forgot to run `brew audit --strict` before committing. My bad.

> Why do we need to replace this test?
We want to make sure that the host machine is handling each of the processes correctly.  This verifies that the architecture/OS/build dependencies are all handling the case soundly.  Things such as endianness, unicode, char signedness, etc should all be confirmed against the target machine (which is not the same as the build machine).

Comment (carlocab) [reply Formula/mpw.rb:46]: Formula test blocks aren't the place for that sort of verification. @SMillerDev also says this here: https://github.com/Homebrew/homebrew-core/pull/71428#discussion_r578593710

The test block is just a sanity check that the build didn't go horribly wrong, and is meant to simulate a user interacting briefly with the software's functionality.

Comment (lhunath) [reply Formula/mpw.rb:46]: mkay — so homebrew doesn't provide any way to allow the user to verify that the software operates reliably in their environment?

I do run integration tests during development, but this only tests the software against the development machine, not the target architecture.  Would you recommend I add a formula flag to build & install the test suite (ie. `mpw-tests`)?

**Comment by lhunath** [comment]
I have completely removed the integration tests.

File: Formula/mpw.rb (hunk 4 above)
Comment (SMillerDev) [reply Formula/mpw.rb:46]: > mkay — so homebrew doesn't provide any way to allow the user to verify that the software operates reliably in their environment?

They can run it to see if it works. Anything else is up to upstream in combination with the simple test in brew.

> I do run integration tests during development, but this only tests the software against the development machine, not the target architecture.

The only advice I can give is: add more CI machines to test for macOS as well.

> Would you recommend I add a formula flag to build & install the test suite (ie. mpw-tests)?

No, we don't allow options in homebrew-core.

Comment (carlocab) [reply Formula/mpw.rb:46]: You can set up GitHub Actions running macOS now. That should provide you with at least one option for running your integration tests on macOS.

GitHub runners also come with `brew` installed, so you can even run your integration tests against the bottles we build.

**Review by carlocab — APPROVED** [review]
Thanks, @lhunath.

**Pull request closed without merging**